components/CheckoutWizard/CheckoutWizard.tsx

```diff
@@ -30,23 +32,41 @@ export const CheckoutWizard: FunctionComponent<Props> = ({ children, assetId })
   const router = useRouter()
   const intl = useIntl()
   const { assetData, isLoading } = useAssetData(assetId)
-  // @TODO We might or might not need this initial value later on
-  // const localCheckoutData = JSON.parse(window.localStorage.getItem("checkout_data") ?? "{}")
+  const { removeItem } = useCheckoutData()
 
   useEffect(() => {
-    router.events.on("routeChangeComplete", () => {
-      const steps = ["terms", "access", "redirect"]
+    const handleRouteChange = () => {
       const stepByPathName = steps.findIndex((step) => router.pathname.includes(step))
       // @TODO If a user manually refresh other steps than number 1, this logic
       // for current step will fail. See issue #83
       setCurrentStep(stepByPathName)
-    })
+    }
+
+    router.events.on("routeChangeComplete", handleRouteChange)
+
+    return () => {
+      router.events.off("routeChangeComplete", handleRouteChange)
+    }
   }, [router])
 
+  useEffect(() => {
+    const handleRouteChange = (url: string) => {
+      const isInTheCheckoutFlow = steps.find((step) => url.includes(step))
```

Comment:
what if the user navigates to `/checkout/request-access` or `/checkout/access/some/nested/route`?

Reply:
You will not loose your data at least 😅  But yes, a little bit hard to define all the bits and pieces of this, I guess it might be better if the algorithm was 
`if has checkout as part of the path`
? Should we do it now or later

Reply:
yeah let's do that :+1: